math/mathcore/test/vectorization/testTMathStatistic.cxx

```diff
@@ -0,0 +1,104 @@
+#ifdef R__HAS_VECCORE
+
+#include "TMath.h"
+#include "TMathVectorized.h"
+
+#include "Math/Random.h"
+#include "TRandom1.h"
+#include "TStopwatch.h"
+
+#include "gtest/gtest.h"
+
+#include <VecCore/VecCore>
+#include <iostream>
+#include <random>
+
+using namespace std;
+using namespace vecCore;
+
+using ROOT::Double_v;
+
+/**
+* [SetUp description]
+*/
+class VectorizationTest : public ::testing::Test {
+protected:
+   VectorizationTest() {
+      // Randomize input data and parameters
+      TRandom1 rndmzr;
+      rndmzr.RndmArray(inputSize * VectorSize<Double_v>(), linearInput);
+      rndmzr.RndmArray(inputSize, mean);
+      rndmzr.RndmArray(inputSize, sigma);
+
+      // Set -100 < mean < 100 and 0 < sigma < 200
+      for (size_t i = 0; i < inputSize; i++) {
+          mean[i] = mean[i] * 200 - 100;
+          sigma[i] *= 200;
+      }
+
+      // Copy input linear data to the vectorized array
+      for (size_t caseIdx = 0; caseIdx < inputSize; caseIdx++) {
+          for (size_t vecIdx = 0; vecIdx < VectorSize<Double_v>(); vecIdx++) {
+             vectorInput[caseIdx][vecIdx] = linearInput[VectorSize<Double_v>() * caseIdx + vecIdx];
+          }
+      }
+
+   }
+
+   virtual ~VectorizationTest() {
+      // You can do clean-up work that doesn't throw exceptions here.
+   }
+
+   // If the constructor and destructor are not enough for setting up
+   // and cleaning up each test, you can define the following methods:
+
+   virtual void SetUp() {
+      // Code here will be called immediately after the constructor (right
+      // before each test).
+   }
+
+   virtual void TearDown() {
+      // Code here will be called immediately after each test (right
+      // before the destructor).
+   }
+
+   // Data available to all tests in this suite
+
+   static const int inputSize = 100000;
+
+   // Vectorized and linear input
+   Double_v vectorInput[inputSize];
+   Double_t linearInput[inputSize * VectorSize<Double_v>()];
+
+   // Vectorized and linear output
+   Double_v vectorOutput[inputSize];
+   Double_t linearOutput[inputSize * VectorSize<Double_v>()];
+
+   // Parameters vector
+   Double_t mean[inputSize];
+   Double_t sigma[inputSize];
+
+};
+
+TEST_F(VectorizationTest, Dummy) {
+   for (size_t caseIdx = 0; caseIdx < inputSize; caseIdx++) {
+      vectorOutput[caseIdx] = TMath::Gaus(vectorInput[caseIdx], mean[caseIdx], sigma[caseIdx]);
+   }
+
+   size_t idx;
+
+   for (size_t caseIdx = 0; caseIdx < inputSize; caseIdx++) {
+      for (size_t vecIdx = 0; vecIdx < VectorSize<Double_v>(); vecIdx++) {
+         idx = caseIdx * VectorSize<Double_v>() + vecIdx;
+         linearOutput[idx] = TMath::Gaus(linearInput[idx], mean[caseIdx], sigma[caseIdx]);
+      }
+   }
+
+   for (size_t caseIdx = 0; caseIdx < inputSize; caseIdx++) {
+      for (size_t vecIdx = 0; vecIdx < VectorSize<Double_v>(); vecIdx++) {
+         ASSERT_NEAR(linearOutput[caseIdx*VectorSize<Double_v>() + vecIdx], vectorOutput[caseIdx][vecIdx], 1e-15);
```

Comment:
Yes, I did click on each commit and comment. I would advise to amend commits and rebase so we don't have commits without meaning in the git history. All commits should contain final changes ideally. Let me know if you need help with that.